sdk/test-utils/recorder-new/src/core-v2-recorder.ts

```diff
@@ -58,45 +59,13 @@ export class TestProxyHttpClient {
         );
       }
       this.sanitizer = new Sanitizer(this.mode, this.url, this.httpClient);
+      this.requestModifier = new RecorderRequestModifier(this.mode, this.url);
     }
     this.variables = {};
   }
 
-  /**
-   * For core-v1 (core-http)
-   */
-  redirectRequest(request: WebResourceLike): void;
-
-  /**
-   * For core-v2 (core-rest-pipeline)
-   */
-  redirectRequest(request: PipelineRequest): void;
-
-  /**
-   * redirectRequest updates the request in record and playback modes to hit the proxy-tool with appropriate headers.
-   * Works for both core-v1 and core-v2
-   */
-  redirectRequest(request: WebResourceLike | PipelineRequest): void {
-    if (isPlaybackMode() || isRecordMode()) {
-      if (!request.headers.get("x-recording-id")) {
-        if (this.recordingId === undefined) {
-          throw new RecorderError("Recording ID must be defined to redirect a request");
-        }
-
-        request.headers.set("x-recording-id", this.recordingId);
-        request.headers.set("x-recording-mode", this.mode);
-
-        const upstreamUrl = new URL(request.url);
-        const redirectedUrl = new URL(request.url);
-        const providedUrl = new URL(this.url);
-
-        redirectedUrl.host = providedUrl.host;
-        redirectedUrl.port = providedUrl.port;
-        redirectedUrl.protocol = providedUrl.protocol;
-        request.headers.set("x-recording-upstream-base-uri", upstreamUrl.toString());
-        request.url = redirectedUrl.toString();
-      }
-    }
+  getRecordingId() {
```

Comment:
could we just use the public property `recordingId`?

Reply:
The problem is I didn't want to pass the whole RecorderClient object to the httpClient of RecorderClientCoreV1,
to pass only what's needed, I had to make the getRecordingId a function like this.

Reply:
I can move this function to RecorderClientCoreV1 in case it is bothering. 🙂

Reply:
removed completely